cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/StorageModule.java

```diff
@@ -48,11 +49,13 @@ protected void configure() {
     bind(StructuredTableAdmin.class).toProvider(StructuredTableAdminProvider.class).in(Scopes.SINGLETON);
     bind(StructuredTableRegistry.class).toProvider(StructuredTableRegistryProvider.class).in(Scopes.SINGLETON);
     bind(DataSourceInstantiator.class).in(Scopes.SINGLETON);
+    bind(DataSource.class).toProvider(DataSourceInstantiator.class).in(Scopes.SINGLETON);
 
     expose(TransactionRunner.class);
     expose(StructuredTableAdmin.class);
     expose(StructuredTableRegistry.class);
     expose(DataSourceInstantiator.class);
```

Comment:
Why expose this class? Typically a provider class should remain as private binding.

Reply:
This was there when this class does not implement Provider, removed it now